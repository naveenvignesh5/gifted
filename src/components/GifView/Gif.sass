@import '../../theme'
    
.gif-wrapper
    display: grid
    grid-template-columns: 1fr 1fr 1fr
    +mobile
        grid-template-columns: 1fr 1fr
