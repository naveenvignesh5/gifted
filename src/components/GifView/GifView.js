import React from "react";

const GifView = ({ src }) => {
  return <img width="200px" height="200px" src={src} alt="animated" />;
};

export default GifView;
