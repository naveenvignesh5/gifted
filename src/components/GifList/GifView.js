import React, { memo, useState } from "react";

import "./Gif.sass";

const GifView = memo(({ src }) => {
  const [imageLoaded, updateImageLoaded] = useState(false);

  return (
    <div className="gif-image-wrapper">
      <div className="gif-loading" style={{ display: imageLoaded ? 'none' : 'flex' }}>Loading GIF...</div>
      {imageLoaded && <div className="gif-image-overlay">
        <a className="download-button" href={src}>View</a>
      </div>}
      <img
        className="gif-image"
        style={{ display: imageLoaded ? 'block' : 'none' }}
        onLoad={() => {
          console.log('image loaded')
          updateImageLoaded(true);
        }}
        src={src}
        alt="animated"
      />
    </div>
  );
});

export default GifView;
