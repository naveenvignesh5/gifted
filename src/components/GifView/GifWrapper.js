import React from "react";
import GifView from "./GifView";

import './Gif.sass';

const GifWrapper = ({ gifs = [] }) => {
  return (
    <div className="gifs-wrapper">
      {gifs.map((gif, index) => (
        <GifView
          key={index.toString()}
          src={`https://media.giphy.com/media/${gif.id}/giphy.gif`}
        />
      ))}
    </div>
  );
};

export default GifWrapper;
