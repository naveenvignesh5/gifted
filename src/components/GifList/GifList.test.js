import React from "react";
import renderer from "react-test-renderer";

import GifView from "./GifView";

describe("GifView", () => {
  it("snapshot renders", () => {
    const component = renderer.create(
      <GifView src="https://media.giphy.com/media/J4CLEVEHB7L1n8uBbW/source.gif" />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
