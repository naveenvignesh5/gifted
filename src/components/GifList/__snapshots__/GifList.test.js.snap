// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifView snapshot renders 1`] = `
<div
  className="gif-image-wrapper"
>
  <div
    className="gif-loading"
  >
    Loading GIF...
  </div>
  <img
    alt="animated"
    className="gif-image static"
    onLoad={[Function]}
  />
  <img
    alt="animated"
    className="gif-image dynamic"
    onLoad={[Function]}
    src="https://media.giphy.com/media/J4CLEVEHB7L1n8uBbW/source.gif"
  />
</div>
`;
